### ISSUE TITLE
[CI] Problems in mixed cluster cleanup in ESRestTestCase#wipeRollupJobs

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#36044]
Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+default-distro-bwc-tests/131/console

This one has a bunch of failures in the mixed cluster tests:

```
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=index/60_refresh/refresh=wait_for waits until changes are visible in search}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=indices.put_template/10_basic/Put multiple template}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=indices.update_aliases/20_routing/Numeric Routing}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=nodes.stats/10_basic/Nodes stats level}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=nodes.stats/11_indices_metrics/Metric - multi}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search.aggregation/40_range/Date Range Missing}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search/160_exists_query/Test exists query on mapped date field}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search/160_exists_query/Test exists query on unmapped geo_point field}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search/30_limits/Docvalues_fields size limit}
org.elasticsearch.backwards.IndexingIT testIndexVersionPropagation
```

The failures look similar to this, so it probably not be the tests itself that fail but some cleanup:
```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:46698], URI [/_rollup/job/_all], status line [HTTP/1.1 404 Not Found]
{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"},"status":404}
	at __randomizedtesting.SeedInfo.seed([3DF3A00DADE684BC:B5A79FD7031AE944]:0)
	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:477)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:407)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:193)
```

I tried to reproduce with one of the reproduce lines on master:
```
./gradlew :qa:mixed-cluster:v6.6.0#mixedClusterTestRunner \
  -Dtests.seed=3DF3A00DADE684BC \
  -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT \
  -Dtests.method="test {p0=search/160_exists_query/Test exists query on mapped date field}" \
  -Dtests.security.manager=true \
  -Dtests.locale=ar \
  -Dtests.timezone=Kwajalein \
  -Dtests.distribution=zip \
  -Dcompiler.java=11 \
  -Druntime.java=8
``` 

And got this:
```
ERROR   7.60s | MixedClusterClientYamlTestSuiteIT.test {p0=search/160_exists_query/Test exists query on mapped date field} <<< FAILURES!
   > Throwable #1: org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:52996], URI [/_rollup/job/_all], status line [HTTP/1.1 404 Not Found]
   > {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"},"status":404}
   >    at __randomizedtesting.SeedInfo.seed([3DF3A00DADE684BC:B5A79FD7031AE944]:0)
   >    at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
   >    at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
   >    at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:477)
   >    at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:407)
   >    at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:193)
   >    at java.lang.Thread.run(Thread.java:748)
   > Caused by: org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:52996], URI [/_rollup/job/_all], status line [HTTP/1.1 404 Not Found]
   > {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"},"status":404}
   >    at org.elasticsearch.client.RestClient$1.completed(RestClient.java:307)
   >    at org.elasticsearch.client.RestClient$1.completed(RestClient.java:292)
   >    at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
   >    at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
   >    at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
   >    at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
   >    at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
   >    at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
   >    at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
   >    at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
   >    at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
   >    at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
   >    at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
   >    at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
   >    at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
   >    at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
   >    ... 1 more
  2> NOTE: leaving temporary files on disk at: /Users/christoph/workspace/elasticsearch/qa/mixed-cluster/build/testrun/v6.6.0#mixedClusterTestRunner/J0/temp/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT_3DF3A00DADE684BC-001
  2> NOTE: test params are: codec=Asserting(Lucene80): {}, docValues:{}, maxPointsInLeafNode=1457, maxMBSortInHeap=7.982721853983626, sim=Asserting(org.apache.lucene.search.similarities.AssertingSimilarity@138084b3), locale=ar, timezone=Kwajalein
  2> NOTE: Mac OS X 10.14.1 x86_64/Oracle Corporation 1.8.0_181 (64-bit)/cpus=8,threads=1,free=445255424,total=514850816
  2> NOTE: All tests run in this JVM: [MixedClusterClientYamlTestSuiteIT]
```

**Comment by cbuescher** [comment]
@polyfractal labeled this "Rollup" because of the "wipeRollupJobs" in the Stacktrace, but I'm not sure this has anything to do with Rollups themselves. Maybe you can shed a light on whats going wrong at teardown though.

**Comment by cbuescher** [comment]
This one looks quiete similar although its other tests that fail: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+default-distro/131/console

```
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=cat.aliases/10_basic/Alias against closed index}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=cat.templates/10_basic/Select columns}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=cat.templates/10_basic/Sort templates}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=index/12_result/Index result field}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=indices.get/10_basic/Missing index should return empty object if ignore_unavailable}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=indices.shard_stores/10_basic/multiple indices test}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=ingest/10_basic/Test Get All Pipelines}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search.aggregation/180_percentiles_tdigest_metric/Filtered test}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=search.aggregation/220_filters_bucket/Bad params}
org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT test {p0=suggest/30_context/Multi contexts should work}
```

One of the failures:
```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:39465], URI [/_rollup/job/_all], status line [HTTP/1.1 404 Not Found]
{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_expression","resource.id":"_rollup","index_uuid":"_na_","index":"_rollup"},"status":404}
	at __randomizedtesting.SeedInfo.seed([638A269756C2ADF8:EBDE194DF83EC000]:0)
	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:477)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:407)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:193)
```

**Comment by polyfractal** [comment]
I think this is the same as https://github.com/elastic/elasticsearch/issues/35979.  I just merged https://github.com/elastic/elasticsearch/pull/36000 which should fix the issue.  Sorry for the noise, forgot to merge that last night.

I'll keep an eye on things and see if the fix worked, then will close this issue if it's fixed.

It fails widely because the `wipeRollup()` method (called against all APIs) needs to use the older `_xpack` endpoint instead of the newer `_rollup` endpoint in mixed clusters.

**Comment by cbuescher** [comment]
@polyfractal great, so my guess that you might know something about this wasn't too far off after all ;-)

**Comment by polyfractal** [comment]
hehe indeed :)  Thanks for the ping!

**Comment by jpountz** [comment]
@polyfractal can it be closed now or is there still something that should be looked into?

**Comment by cbuescher** [comment]
@polyfractal I got new failures that look related but somehow different:
```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:32927], URI [/_xpack/rollup/job/_all], status line [HTTP/1.1 200 OK]
Warnings: [[GET /_xpack/rollup/job/{id}/] is deprecated! Use [GET /_rollup/job/{id}] instead.]
{"jobs":[]}
	at __randomizedtesting.SeedInfo.seed([CAB0377480BB810C:4B236354254D4753]:0)
	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:687)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeRollupJobs(ESRestTestCase.java:479)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:407)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:193)
```
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+default-distro-bwc-tests/132/console

Would that be covered by #36000 as well?

**Comment by polyfractal** [comment]
It's related, although technically a different failure.  I'm working on it although it's proving tricky to fix.

Basically, the alternate case is now failing (where we use the deprecated endpoint on the v7 node).  The test's `adminClient` is built in strict mode, so we can't catch the deprecations.  And we don't know which node the client is going to talk to, so we can't tailor the request.

I'm investigating if it's possible to add a "homogeneous" admin client which only talks to a single node version, or perhaps a "single node" admin client which only talks to one (known) node.

We could also just use a non-strict admin client but that seems like a worse idea because we could miss unintentional warnings.

/cc @jasontedor in case you have any ideas :)

**Comment by jasontedor** [comment]
I wonder if we should amend the test runner to excuse deprecation warnings for `_xpack` endpoints.

**Comment by polyfractal** [comment]
Update:  we decided to revert the fix-that-didn't-fix (40c54454805a468c2bc5d02073044addb68e8c21) and the `/_xpack/rollup/` deprecation (61c2db5ebbec9b32a02be79a8adce4728295b207) until we can sort out the testing situation.  The builds should quiet back down now.

**Comment by nik9000** [comment]
I'm using this to track my un-revert progress.
